### PR TITLE
Update documentation of `toFloat32/64` and variants

### PR DESCRIPTION
**Pull request by Blargian**
Updates documentation for `toFloat32/64` and their variants as part of [#2022](https://github.com/ClickHouse/clickhouse-docs/issues/2022) 

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Comment by Blargian** [comment]
@rschu1ze tagging for review

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 1]
```diff
@@ -3045,13 +3045,425 @@ toUInt256OrDefault('abc', CAST('0', 'UInt256')):  0
...
+
+**Arguments**
+
+- `expr` — Expression returning a number or a string representation of a number. [Expression](../syntax.md/#syntax-expressions).
+
+Supported arguments:
+- Values of type (U)Int8/16/32/64/128/256.
+- String representations of (U)Int8/16/32/128/256.
+- Values of type Float32/64, including `NaN` and `Inf`.
+- String representations of Float32/64, including `NaN` and `Inf`.
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/type-conversion-functions.md:3066]: String representations of Float32/64, including `NaN` and `Inf` (case-insensitive).

File: docs/en/sql-reference/functions/type-conversion-functions.md [hunk 2]
```diff
@@ -3045,13 +3045,425 @@ toUInt256OrDefault('abc', CAST('0', 'UInt256')):  0
...
+**Example**
+
+Query:
+
+```sql
+SELECT
+    toFloat32(42.7),
+    toFloat32('42.7'),
+    toFloat32('NaN')
+FORMAT vertical;
```
Comment (rschu1ze) [inline docs/en/sql-reference/functions/type-conversion-functions.md:3084]: The format name recently became case-insensitive but somehow I still like uppercase `Vertical` 3% better.

Comment (Blargian) [reply docs/en/sql-reference/functions/type-conversion-functions.md]: @rschu1ze I can tell that you are German :-)

Comment (rschu1ze) [reply docs/en/sql-reference/functions/type-conversion-functions.md]: 103% true.

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 0bb8d1a8ede4906a1cdc75af2dbf1e190ff355fc with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67908/0bb8d1a8ede4906a1cdc75af2dbf1e190ff355fc/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/67908/0bb8d1a8ede4906a1cdc75af2dbf1e190ff355fc/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

**Pull request merged**